packages/host/app/components/card-catalog-modal.gts

```diff
@@ -54,30 +63,84 @@ export default class CardCatalogModal extends Component<Signature> {
                   data-test-create-new
                 >Create New</Button>
               {{/if}}
+              <div class='card-catalog__results-length'>{{if
+                  this.currentRequest.search.instances.length
+                  this.currentRequest.search.instances.length
+                  '0'
+                }}
+                results</div>
               <ul class='card-catalog' data-test-card-catalog>
                 {{#each this.currentRequest.search.instances as |card|}}
-                  <li data-test-card-catalog-item={{card.id}}>
+                  <li
+                    class={{cn
+                      'card-catalog-item'
+                      is-selected=(eq this.selectedCard.id card.id)
+                    }}
+                    data-test-card-catalog-item={{card.id}}
+                  >
                     <Preview
                       @card={{card}}
                       @format='embedded'
                       @context={{@context}}
                     />
                     <button
-                      {{on 'click' (fn this.pick card)}}
+                      class='card-catalog-item__select'
+                      {{on 'click' (fn this.select card)}}
                       data-test-select={{card.id}}
-                    >
-                      Select
-                    </button>
+                      aria-label='Select'
+                    />
+                    <IconButton
+                      class='card-catalog-item__hover-button card-catalog-item__preview'
+                      @icon='eye'
+                      aria-label='preview'
+                    />
+                    <IconButton
+                      class='card-catalog-item__hover-button card-catalog-item__more-actions'
+                      @icon='more-actions'
+                      aria-label='more actions'
+                    />
                   </li>
                 {{else}}
                   <p>No cards available</p>
                 {{/each}}
               </ul>
             {{/if}}
           </div>
+          <footer class='dialog-box__footer card-catalog-modal__footer'>
+            <label class='card-catalog-modal__url-search'>
+              <span>Enter Card URL:</span>
+              <BoxelInput @value={{this.cardURL}} placeholder='http://' />
+            </label>
+            <div>
+              <Button
+                @kind='secondary-light'
+                @size='tall'
+                @disabled={{eq this.selectedCard undefined}}
```

Comment:
It seems like the Cancel button should still be available even if no card has been selected.